src/backend/gpopt/translate/CTranslatorDXLToPlStmt.cpp

```diff
@@ -3608,6 +3608,11 @@ CTranslatorDXLToPlStmt::TranslateDXLCTEProducerToSharedScan(
 		child_plan = materialize_plan;
 	}
 
+	// targetlist mismatch leads to different tuple bindings, see #12796
+	if (list_length(child_plan->targetlist) != list_length(plan->targetlist))
```

Comment:
I don't see any valid situation when parent node may has, for example `i, j, k` columns, when child node has `i, j, l` columns or wrong sorted columns. If there are some situations when one set does not include the other it should be hooked and fixed somewhere outside `TranslateDXLCTEProducerToSharedScan` function.

Reply:
OK then point in comment that in current plan node we expect target list as sub-list from child one because it's not evident that comparing sizes we compare lists.